src/libsyntax/parse/parser.rs

```diff
@@ -7263,11 +7268,9 @@ impl<'a> Parser<'a> {
         }
         if self.check_keyword(kw::Impl) ||
            self.check_keyword(kw::Unsafe) &&
-                self.look_ahead(1, |t| t.is_keyword(kw::Impl)) ||
-           self.check_keyword(kw::Default) &&
-                self.look_ahead(1, |t| t.is_keyword(kw::Impl)) ||
```

Comment:
side-note (pre existing): `default unsafe fn` is eaten by this `if` and will error?

Reply:
Yeah I believe so.